runners/apex/src/main/java/org/apache/beam/runners/apex/translators/functions/ApexGroupByKeyOperator.java

```diff
@@ -97,8 +96,8 @@
   @Bind(JavaSerializer.class)
   private final SerializablePipelineOptions serializedOptions;
   @Bind(JavaSerializer.class)
-// TODO: InMemoryStateInternals not serializable
-  private transient Map<ByteBuffer, StateInternals<K>> perKeyStateInternals = new HashMap<>();
+  private final StateInternalsFactory<K> stateInternalsFactory;
+  private Map<ByteBuffer, StateInternals<K>> perKeyStateInternals = new HashMap<>();
```

Comment:
Technically, though, you shouldn't need to serialize the `StateInternals`. The idea is that the `StateInternalsFactory` represents the worker-side capability to create a `StateInternals`. So the `StateInternals` should be created fresh on each worker anyhow.